### PR TITLE
[FIX] stock: immediate transfers in the future

### PR DESCRIPTION
**Pull request by Whenrow**
Immediate transfers only fill the quantity done of stock moves/stock move lines
Those transfer are thus not taken into account for the forecast computation if they
are not validated. Indeed, the stock quant are not yet updated and the initial demand
is left blank.

This commit duplicate the quantity done of immediate transfer on their initial demand
this quantity is still hidden in the view but can now enter in the forecast quantity.

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by sle-odoo** [comment]
@amoyaux

File: addons/stock/models/stock_move_line.py [hunk 1]
```diff
@@ -199,6 +199,12 @@ def create(self, vals_list):
         mls = super(StockMoveLine, self).create(vals_list)
 
         for ml in mls:
+            if ml.move_id and \
+                    ml.move_id.picking_id and \
+                    ml.move_id.picking_id.immediate_transfer and \
+                    ml.move_id.state != 'done' and \
+                    'qty_done' in vals:
+                ml.move_id.product_uom_qty = ml.move_id.quantity_done
```
Comment (fmdl) [inline addons/stock/models/stock_move_line.py]: move_id.product_uom_qty will be write multi time.

replace by:
```python
if 'qty_done' in vals:
     moves = self.filtered(lambda ml: ml.move_id.state != 'done' and ml.move_id.picking_id and ml.move_id.picking_id.immediate_transfer).mapped('move_id')
     for move in moves:
          move.product_uom_qty = move.quantity_done
```

Comment (Whenrow) [reply addons/stock/models/stock_move_line.py]: first condition has 'ml.move_id.state != 'done' and second ml.state == 'done' which is complementary

**Comment by sle-odoo** [comment]
robodoo r+

File: addons/stock/models/stock_move_line.py (hunk 1 above)
Comment (fmdl) [reply addons/stock/models/stock_move_line.py]: but if there are multi ml per move you write the same information multi time.

**Pull request closed without merging**